.expected-plugins/centos9

```diff
@@ -0,0 +1,113 @@
+# CentOS 9
+libcollectdclient
+apache
+apcups
+battery
+bind
+buddyinfo
+capabilities
+ceph
+cgroups
+chrony
+connectivity
+conntrack
+contextswitch
+cpu
+cpufreq
+cpusleep
+curl
+curl_json
+dbi
```

Comment:
Other lists have also `curl_xml`.

Reply:
I have created these lists in slightly different ways. For CentOS 9, I have created an empty file, and then imported all the plugins there were built by CirrusCI. For Debian 12, I have imported the list from `main` and then commented out all the non-working plugins. The *curl_xml plugin* is one that is commented out.

Reply:
Any thoughts on those files listing only plugins that should NOT be tested?

It should be a much smaller set, and therefore much easier to review & maintain in future...

Reply:
An inverted list wouldn't provide the same value IMHO. The value this system provides is that it alerts us to the fact if a plugin that used to build on, say, Debian Unstable stops being built there.

With an inverted list, we would have a list of plugins that must not be built. This would trigger when a contributor ports a plugin or migrates a plugin to a more modern dependency. These are not really cases we need to be *alerted* on.

I do see the point about maintenance, though. We could expand the check script to accept multiple files. We could use that to move all the "will build on all Linux distros" plugins to a central location and only maintain lists of plugins in which the distros differ. What do you think?

Reply:
> I do see the point about maintenance, though. We could expand the check script to accept multiple files. We could use that to move all the "will build on all Linux distros" plugins to a central location and only maintain lists of plugins in which the distros differ. What do you think?

I was thinking that there would a common "do-build" list, and per-distro "do-not-build" _exceptions_ for plugins on that list:
`grep -v -f $DISTRO-exceptions.txt all-plugins-to-build.txt > plugins-to-build-on-$DISTRO.txt`

In general same plugins should be buildable on all distros, right?  So exceptions should be really rare.

Having explicit  "do-not-build" lists would IMHO both make the lists easier to maintain, **and** very clearly document what those distro specific exceptions are.


> With an inverted list, we would have a list of plugins that must not be built. This would trigger when a contributor ports a plugin or migrates a plugin to a more modern dependency. These are not really cases we need to be *alerted* on.

Unless new plugin is added to common "do-build" list, it would not be built.

Reply:
I think this is a very similar approach. The important aspect (to me) is that it breaks out a "base line", with per-distro specialization. Whether that specialization is an allow or deny list is not a huge difference IMHO.

There is a rather big difference between operating systems. Many plugins should build on any Linux distro, but won't build on FreeBSD. That's why I thought composition would be a good way of reducing duplication while allowing us to have different base lines for Linux and FreeBSD, for example.

Reply:
> There is a rather big difference between operating systems. Many plugins should build on any Linux distro, but won't build on FreeBSD.

There are more Linux distros in CI than the others, so I would think per-distro deny lists deducted from "all-plugins", would in total be less lines than per-distro allow lists added to "base-plugins" one, but I have no idea how significant the difference is...